cpp/periodic_module/periodic_iterate.cpp

```diff
@@ -0,0 +1,306 @@
+
+    for (size_t i = 0; i < param_row_size; i++) {
+      if (batch[row][i].type() == mg::Value::Type::Node) {
+        constructed_row.Insert(columns[i], mg::Value(static_cast<int64_t>(batch[row][i].ValueNode().id().AsInt())));
+      } else if (batch[row][i].type() == mg::Value::Type::Relationship) {
+        constructed_row.Insert(columns[i],
+                               mg::Value(static_cast<int64_t>(batch[row][i].ValueRelationship().id().AsInt())));
+      } else {
+        constructed_row.Insert(columns[i], batch[row][i]);
+      }
+    }
+
+    list_value.Append(mg::Value(std::move(constructed_row)));
+  }
+
+  params.Insert(kBatchInternalName, mg::Value(std::move(list_value)));
+
+  return params;
+}
+
+auto ConstructFinalQuery(const std::string &running_query, const std::string &prefix_query) {
+  return fmt::format("{} {}", prefix_query, running_query);
+}
+
+void ExecuteRunningQuery(const std::string running_query, const std::vector<std::string> &columns,
+                         const std::vector<std::vector<mg::Value>> &batch) {
+  if (!batch.size()) {
+    return;
+  }
+
+  auto param_names = ExtractParamNames(columns, batch[0]);
+  auto prefix_query = ConstructQueryPrefix(param_names);
+  auto final_query = ConstructFinalQuery(running_query, prefix_query);
+
+  auto query_params = ConstructQueryParams(columns, batch);
+
+  mg::Client::Params session_params{.host = "localhost", .port = 7687};
+  auto client = mg::Client::Connect(session_params);
+  if (!client) {
+    throw std::runtime_error("Unable to connect to client!");
+  }
+  if (!client->Execute(final_query, query_params.AsConstMap())) {
+    throw std::runtime_error("Error while executing periodic iterate!");
+  }
+
+  client->DiscardAll();
+}
+
+void ValidateBatchSize(const mgp::Value &batch_size_value) {
+  if (batch_size_value.IsNull()) {
+    throw std::runtime_error(fmt::format("Configuration parameter {} is not set.", kConfigKeyBatchSize));
+  }
+  if (!batch_size_value.IsInt()) {
+    throw std::runtime_error("Batch size not provided as an integer in the periodic iterate configuration!");
+  }
+
+  const auto batch_size = batch_size_value.ValueInt();
+  if (batch_size <= 0) {
+    throw std::runtime_error("Batch size must be a non-negative number!");
+  }
+}
+
+mg::Client::Params GetClientParams() {
+  auto *host = kDefaultHost;
+  auto port = kDefaultPort;
+  auto *username = "";
+  auto *password = "";
+
+  auto *maybe_host = std::getenv(kMgHost);
+  if (maybe_host) {
+    host = std::move(maybe_host);
+  }
+
+  const auto *maybe_port = std::getenv(kMgPort);
+  if (maybe_port) {
+    port = static_cast<uint16_t>(std::move(*maybe_port));
+  }
+
+  const auto *maybe_username = std::getenv(kMgUsername);
+  if (maybe_username) {
+    username = std::move(maybe_username);
+  }
+
+  const auto *maybe_password = std::getenv(kMgPassword);
+  if (maybe_password) {
+    password = std::move(maybe_password);
+  }
+
+  return mg::Client::Params{.host = std::move(host),
```

Comment:
just curious, what happens if some of these env variables aren't set, how will std::move behave?

Reply:
I think @Josipmrden does the correct check, gets env and if doesn't exist it should be predefined.

Reply:
I initialized it all to default values in the beginning, so I guess it's safe to assume that it will move the default variables since it did not go to if